parser/parser.go

```diff
@@ -40,10 +41,11 @@ type Test struct {
 }
 
 var (
-	regexStatus   = regexp.MustCompile(`^\s*--- (PASS|FAIL|SKIP): (.+) \((\d+\.\d+)(?: seconds|s)\)$`)
-	regexCoverage = regexp.MustCompile(`^coverage:\s+(\d+\.\d+)%\s+of\s+statements$`)
-	regexResult   = regexp.MustCompile(`^(ok|FAIL)\s+(.+)\s(\d+\.\d+)s(?:\s+coverage:\s+(\d+\.\d+)%\s+of\s+statements)?$`)
-	regexOutput   = regexp.MustCompile(`(    )*\t(.*)`)
+	regexStatus       = regexp.MustCompile(`^\s*--- (PASS|FAIL|SKIP): (.+) \((\d+\.\d+)(?: seconds|s)\)$`)
+	regexCoverage     = regexp.MustCompile(`^coverage:\s+(\d+\.\d+)%\s+of\s+statements$`)
+	regexResult       = regexp.MustCompile(`^(ok|FAIL)\s+(.+)\s(\d+\.\d+)s(?:\s+coverage:\s+(\d+\.\d+)%\s+of\s+statements)?$`)
+	regexBuildFailure = regexp.MustCompile(`^FAIL\s+(.+)\s\[build failed]$`)
```

Comment:
This can be combined with `regexResult`. Something like this could work:
```regex
^(ok|FAIL)\s+([^ ]+)\s+(?:(\d+\.\d+)s|(\[build failed\]))(?:\s+coverage:\s+(\d+\.\d+)%\sof\sstatements)?$
```